share/spack/spack-completion.bash

```diff
@@ -288,7 +288,7 @@ function _spack_debug {
     fi
 }
 
-function _spack_create-db-tarball {
+function _spack_create_db_tarball {
```

Comment:
Hmm... I don't get that on my CentOS 6 box. What shell are you using? What version?